plugins/catalog-backend/src/processing/ProcessorOutputCollector.ts

```diff
@@ -64,9 +75,9 @@ export class ProcessorOutputCollector {
     };
   }
 
-  private receive(i: CatalogProcessorResult) {
+  private receive(logger: Logger, i: CatalogProcessorResult) {
     if (this.done) {
-      this.logger.warn(
+      logger.warn(
```

Comment:
what's the rationale behind injecting a logger vs using the instanced logger? naming in logs?

Reply:
I want to move towards actually leveraging the structured logging a bit more. In this case, it was because it was my opportunity to signal to the person reading the log _which processor_ that was responsible for this "bad" behavior, making it easier to address.